crates/worker/src/partition/services/non_deterministic/remote_context.rs

```diff
@@ -503,6 +500,43 @@ impl<'a, State: StateReader> InvocationContext<'a, State> {
             &InvocationStatus::Done(get_result_response.clone()),
         )?;
 
+        // Cleanup the journal
+        if let Some((_, journal_meta)) = self.load_journal_metadata(&journal_service_id).await? {
+            self.drop_journal(self.journal_service_id(), journal_meta.length)
+        }
+
+        // Close pending recv and get result
+        self.close_pending_recv(recv_response::Response::Messages(Bytes::new()))
+            .await?;
+        self.close_pending_get_result(get_result_response).await?;
+
+        // Schedule the cleanup
+        self.schedule_cleanup(operation_id, expiry_time, virtual_journal_length);
+
+        Ok(())
```

Comment:
This block looks very similar to what the `kill_invocation` method does after generating the `GetResultResponse`. Could this logic be shared in order to avoid divergence in the future?